python/cudf/cudf/core/column/column.py

```diff
@@ -812,6 +818,19 @@ def values_host(self) -> np.ndarray:
         if len(self) == 0:
             return np.array([], dtype=self.dtype)
 
+        if cudf.get_option(
+            "mode.pandas_compatible"
+        ) and is_pandas_nullable_extension_dtype(self.dtype):
+            if self.dtype.kind in "iuf":
+                with acquire_spill_lock():
+                    res = self.data_array_view(mode="read").copy_to_host()
+                if self.has_nulls():
+                    res = np.where(
+                        self.isnull().values_host,
+                        np.nan,
+                        res,
+                    )
```

Comment:
Should this be done on the GPU instead before copying to host?

Reply:
Done. Moved it to GPU.